pkg/sdn/plugin/controller.go

```diff
@@ -339,7 +340,12 @@ func (plugin *OsdnNode) SetupSDN(localSubnetCIDR, clusterNetworkCIDR, servicesNe
 func (plugin *OsdnNode) SetupEgressNetworkPolicy() error {
 	policies, err := plugin.registry.GetEgressNetworkPolicies()
 	if err != nil {
-		return fmt.Errorf("Could not get EgressNetworkPolicies: %s", err)
+		if kapierrs.IsForbidden(err) {
```

Comment:
Technically that's a different problem - the policy is central ACL in this
case.  We're trying to remove per node config in general, so SDN for
example would be a DaemonSet running alongside the Kubelet, and it would
potentially have DaemonSet compatible mechanisms to solve that.

On Aug 11, 2016, at 1:10 PM, David Eads <notifications@github.com> wrote:

In pkg/sdn/plugin/controller.go
<https://github.com/openshift/origin/pull/10358#discussion_r74462577>:

> @@ -339,7 +340,12 @@ func (plugin *OsdnNode) SetupSDN(localSubnetCIDR, clusterNetworkCIDR, servicesNe
>  func (plugin *OsdnNode) SetupEgressNetworkPolicy() error {
>  	policies, err := plugin.registry.GetEgressNetworkPolicies()
>  	if err != nil {
> -		return fmt.Errorf("Could not get EgressNetworkPolicies: %s", err)
> +		if kapierrs.IsForbidden(err) {

It looks like nodes don't have permission to set labels on themselves. It
could mark itself not Ready (or we could add a new NodeConditionType). But
is that better than erroring out (particularly given that you'd still need
to systemctl restart it after fixing the roles)?

I'd bet that nodes/status keeps labels. I think that not erroring out could
be ok, but its important for us to be able to answer, "which nodes aren't
enforcing these policies". I don't see how I can do that today without
inspecting the logs for each node.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/10358/files/2dab6c5be71b60206fc6afad74c7b6f9b23c3d38#r74462577>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/ABG_p_vIcj9B3Tb0rfFBe94rQqBuS5oTks5qe1eggaJpZM4JiPJL>
.